packages/builder/src/lib/account.ts

```diff
@@ -0,0 +1,46 @@
+import { Account, mnemonicToSecretKey } from "algosdk";
+import * as fs from "fs";
+import YAML from "yaml";
+
+import { BuilderError } from "../internal/core/errors";
+import { ERRORS } from "../internal/core/errors-list";
+import type { AccountDef, HDAccount, MnemonicAccount } from "../types";
+
+export function mkAccounts (input: AccountDef[]): Account[] {
+  const accounts: Account[] = [];
+  for (const i of input) {
+    if ((i as HDAccount).path) {
+      throw new BuilderError(ERRORS.ACCOUNT.HD_ACCOUNT, {});
+    } else if ((i as Account).sk) {
+      accounts.push(i as Account);
+    } else {
+      const ia = i as MnemonicAccount;
+      const a = parseMnemonic(ia.mnemonic);
+      if (a.addr !== ia.addr && ia.addr !== "") {
```

Comment:
Two layers of braces are already a limit for me. Can we remove the `else { __ }` wrapper or split into methods?
You simply match for object type.
Isn't it possible to use `switch` here?
Or `if+continue` without `else`? This would result in less braces.

Reply:
`switch` doesn't work here. I think `if` is the idiomatic way. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
Check out this files for example, `if else ...` chain is used pretty often: 
+ `algosdk/src/transaction.js`
+ lodash.js
....

Reply:
It could help to have `return method( __ )` for nested methods like these.
If you think it's a good way I simply complained about the nested `if`. If you think it's ok then we can have it.

Reply:
I agree - if the function will be bigger, currently it's only few lines of code only for the one case.

Reply:
anyway - adding new method to and using it to do simple push:

```
    } else {
      accounts.push(fromMnemonic(i as MnemonicAccount));
    }
```